osu.Game/Overlays/Direct/DirectListPanel.cs

```diff
@@ -104,53 +107,69 @@ private void load(LocalisationEngine localisation, OsuColour colours)
                                     AutoSizeAxes = Axes.Both,
-                                    Direction = FillDirection.Horizontal,
-                                    Children = new[]
+                                    Direction = FillDirection.Vertical,
+                                    Children = new Drawable[]
                                     {
-                                        new OsuSpriteText
+                                        new Statistic(FontAwesome.fa_play_circle, SetInfo.OnlineInfo?.PlayCount ?? 0)
                                         {
-                                            Text = "mapped by ",
-                                            TextSize = 14,
+                                            Margin = new MarginPadding { Right = 1 },
+                                        },
+                                        new Statistic(FontAwesome.fa_heart, SetInfo.OnlineInfo?.FavouriteCount ?? 0),
+                                        new FillFlowContainer
+                                        {
+                                            Anchor = Anchor.TopRight,
+                                            Origin = Anchor.TopRight,
+                                            AutoSizeAxes = Axes.Both,
+                                            Direction = FillDirection.Horizontal,
+                                            Children = new[]
+                                            {
+                                                new OsuSpriteText
+                                                {
+                                                    Text = "mapped by ",
+                                                    TextSize = 14,
+                                                },
+                                                new OsuSpriteText
+                                                {
+                                                    Text = SetInfo.Metadata.Author.Username,
+                                                    TextSize = 14,
+                                                    Font = @"Exo2.0-SemiBoldItalic",
+                                                },
+                                            },
                                         },
                                         new OsuSpriteText
                                         {
-                                            Text = SetInfo.Metadata.Author.Username,
+                                            Text = $"from {SetInfo.Metadata.Source}",
+                                            Anchor = Anchor.TopRight,
+                                            Origin = Anchor.TopRight,
                                             TextSize = 14,
-                                            Font = @"Exo2.0-SemiBoldItalic",
+                                            Alpha = string.IsNullOrEmpty(SetInfo.Metadata.Source) ? 0f : 1f,
```

Comment:
Would it be viable to not add the `OsuSpriteText` at all if nothing is available to display? I think `BeatmapInfoWedge` does something like that already.

Reply:
That's not really relevant to this PR, the diff just shows it weird.